### PR TITLE
Add Jenkinsfile

### PR DESCRIPTION
**Pull request by alexmuller**
Our new Jenkins builds using a Jenkinsfile

- Remove gh-status, this happens automatically now
- Publish the gem based on the branch name, not an environment variable

## Todo before merge

- [x] Switch over webhook in GitHub repo settings and run branch indexing in Jenkins

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      govuk.cleanupGit()
+      govuk.mergeMasterBranch()
+    }
+
+    stage("Build") {
+      sh "${WORKSPACE}/jenkins.sh"
+    }
+
+    stage("Publish gem") {
+      if (env.BRANCH_NAME == 'master') {
```
Comment (cbaines) [inline Jenkinsfile]: Could this be outside of the stage, so that the stage does not appear to be run when building the branches for Pull Requests?

**Comment by cbaines** [comment]
Is it ok to use the new CI system to build this branch? I think this is what has been done for many other similar branches.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        sh "bundle exec rake publish_gem --trace"
```
Comment (cbaines) [inline Jenkinsfile]: This won't work until https://github.com/alphagov/govuk-puppet/pull/5325 is merged and deployed. Doesn't necessarily mean this branch can't be merged though, just something to be aware of.

**Comment by alexmuller** [comment]
Updated to address your comments, thanks!

**Pull request merged**